### ISSUE TITLE
👀 Aggregate Design: Using Invariants as a Guide

### ISSUE THREAD

**Issue by danielmackay** [danielmackay/dotnet-ef-domain-driven-design#130]
https://www.youtube.com/watch?v=64ngP-aUYPc

**Comment by danielmackay** [comment]
![Image](https://github.com/danielmackay/dotnet-ef-domain-driven-design/assets/2636640/133c91b1-a006-4f7b-9efc-d3920f8584c6)

Thinking about aggregates
- Can the stops (Delivery or Pickups) exist without the shipment?
- Aggregates inforce invariants
- Invariants are business rules that must always be consistent
- If a relationship between two entities must be enforced, then they must be part of an aggregate
- In other words, aggregates can be used to enforce invariants that span across multiple objects
- The recommendation is to always be in a consistent and valid state
